### PR TITLE
Merge C extension beta into main branch.

### PR DESCRIPTION
**Pull request by cheqianh**
Merge C extension into Master branch.

Fixes some conflicts, create a PR to show the success of CI/CD workflow.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by cheqianh** [comment]
Fixes some conflict, create a PR to show succeed of CI/CD workflow.

**Comment by cheqianh** [comment]
Convert it to draft in favor of a pure python implementation release.

File: README.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ This package is designed to work with **Python 3.6+**
 
 Start with the [simpleion](https://ion-python.readthedocs.io/en/latest/amazon.ion.html#module-amazon.ion.simpleion)
 module, which provides four APIs (`dump`, `dumps`, `load`, `loads`) that will be familiar to users of Python's
-built-in JSON parsing module.
+built-in JSON parsing module. Simpleion module's performance is improved by an optional [C extension](https://github.com/amzn/ion-python/blob/c-extension-beta/C_EXTENSION.md).
```
Comment (tgregg) [inline README.md:14]: This link should be updated.

File: C_EXTENSION.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
+# Ion Python C Extension Beta
+
+### ***This feature is considered experimental, in Beta stage.***
```
Comment (tgregg) [inline C_EXTENSION.md:3]: Are we still going to call this "Beta"?

File: C_EXTENSION.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+For symbol `abc` with two annotations `annot1` and `annot2`, `annot1::annot2::abc`:
+```text
+Ion C text        ion_symbol_table::{         symbols:[    "abc", "annot1", "annot2"]}          annot1($11)::annot2($12)::abc($10)
+Ion C binary      \xee\x99\x81\x83  \xde\x95  \x87\xbe\x92 \x83abc\x86annot1\x86annot2 \xe5\x82 \x8b         \x8c         \x71\x0a
+Ion Python binary ion_symbol_table::{         symbols:[    "annot1", "annot2", "abc",]}         annot1($10)::annot2($11)::abc($12)
+ion Python        \xee\x99\x81\x83  \xde\x95  \x87\xbe\x92 \x86annot1\x86annot2\x83abc \xe5\x82 \x8a         \x8b         \x71\x0c
+```
+
+### 2. Known Issues
+1. Type translation errors causing python to crash. Refer to [amzn/ion-python#151](https://github.com/amzn/ion-python/issues/151) for details.
```
Comment (tgregg) [inline C_EXTENSION.md:115]: Is this still an issue?

File: README.md (hunk 1 above)
Comment (cheqianh) [reply README.md:14]: Updated.

File: C_EXTENSION.md (hunk 2 above)
Comment (cheqianh) [reply C_EXTENSION.md:3]: Changed.

File: C_EXTENSION.md (hunk 3 above)
Comment (cheqianh) [reply C_EXTENSION.md:115]: Thanks, I barely see this issue now. I also closed #151 earlier. I'll update the document.

**Comment by cheqianh** [comment]
I created a PR fixing above comments here: https://github.com/amzn/ion-python/pull/183.

**Review by tgregg — APPROVED** [review]

**Pull request merged**